src/main/java/net/minecraftforge/fml/common/registry/VillagerRegistry.java

```diff
@@ -224,30 +235,66 @@ public int getRandomCareer(Random rand)
         private VillagerProfession profession;
         private String name;
         private int id;
-        private ITradeList[][] trades;
+        private List<List<ITradeList>> trades;
 
         public VillagerCareer(VillagerProfession parent, String name)
         {
             this.profession = parent;
             this.name = name;
+            this.trades = new ArrayList<List<ITradeList>>();
             parent.register(this);
         }
 
         public String getName()
         {
             return this.name;
         }
+        
+        public String getUnlocalizedName()
+        {
+            ResourceLocation professionName = profession.getRegistryName();
+            if(professionName.getResourceDomain().equals("minecraft"))
+                return String.format("entity.Villager.%s", this.getName());
+            else
+                return String.format("entity.Villager.%s.%s", professionName.getResourceDomain(), this.getName());
+        }
 
         public ITradeList[][] getTrades()
         {
-            return this.trades;
+            ITradeList[][] returnValue = new ITradeList[this.trades.size()][];
+            for(int i = 0; i < this.trades.size(); i++)
+            {
+               returnValue[i] = this.trades.get(i).toArray(new ITradeList[]{});
```

Comment:
Same thing, don't do it like this.
Instead do:
`trades.get(i).toArray(new ITradeList[trades.get(i).size()])`
Yo also can move the result of `trades.get(i)` into a local variable
```
List<ITradeList> trades = this.trades.get(i);
returnValue[i] = trades.toArray(new ITradeList[trades.size()]);
```